src/MapViewController.swift

```diff
@@ -75,6 +122,46 @@ open class MapViewController: UIViewController, LocationManagerDelegate, TGRecog
       return tgViewController.tilt
     }
   }
+  
+  open var panEnabled = true
+  open var pinchEnabled = true
+  open var rotateEnabled = true
+  open var shoveEnabled = true
+  
+  weak open var singleTapGestureDelegate: MapSingleTapGestureDelegate?
+  weak open var doubleTapGestureDelegate: MapDoubleTapGestureDelegate?
+  weak open var longPressGestureDelegate: MapLongPressGestureDelegate?
+  weak open var panDelegate: MapPanGestureDelegate?
+  weak open var pinchDelegate: MapPinchGestureDelegate?
+  weak open var rotateDelegate: MapRotateGestureDelegate?
+  weak open var shoveDelegate: MapShoveGestureDelegate?
+
+  weak open var featureSelectDelegate: MapFeatureSelectDelegate? {
+    didSet {
+      tgViewController.mapViewDelegate = mapViewDelegatesExist() ? self : nil
```

Comment:
Hmmmmmm I'm unsure this works the way you think it should. If I'm understanding the implementation correctly, this would return `true` if *any* delegate OR `onSceneLoaded` is non-nil? Seems like you'd quickly get out of line with what is nil versus what you've subscribed to listen for events for.

A couple thoughts: it might make for more straightforward code (and less state machine management) if we subscribe to all of Tangram's protocols at the beginning of the lifecycle of MapViewController. It might seem like having Tangram making unnecessary function calls, but once the obj-c runtime caches the IMP map and knows the dispatch pathway between the two objects, it is *very fast* to execute those callbacks. If you're trying to be efficient here by avoiding unnecessary linkages, don't optimize for that. At least, not yet :) 

However, if you want to have this state machine going here, there's an easier way to go about it. `didSet` is guaranteed to be called after the property has been stored, so you could just check the value of the property as it stands now. So, for this, it would be something like `tgViewController.mapViewDelegate = featureSelectDelegate ? self : nil`